### DISCUSSION TITLE
Change database config during runtime?

### DISCUSSION THREAD

**Discussion by che-122** [adonisjs/core#2811] [Help]
Hello,

I've been looking into AdonisJS for a SaaS product. I've run up against an issue where I would like to have the ability to change the user which connects to the DB.

I'm trying to create a tenant aware application. My approach is to enable row level security on the postgres db and let each tenant have full access to the tables. The DB is then aware of what rows on the table each postgres user has access to.

Every time a new tenant is created in my database, I dispatch an event to a class that creates a new user/password combo on the database.

```
public static async create(tenantId: string, tenantSecret: string) {
    await Database.knexRawQuery(`CREATE USER ?? WITH ENCRYPTED PASSWORD 'somesecret';`, [tenantId])
    await Database.knexRawQuery(`GRANT ALL PRIVILEGES ON TABLE users TO ??;`, [tenantId])
    Logger.info(`Granted priveleges to users table for tenant "${tenantId}"`)
  }
```

The problem here is I would need to make sure each tenant logs in and uses their own username & password for the database. So the database connection needs to be dynamic at runtime.

I've been looking at this https://docs.adonisjs.com/reference/database/query-client but I can't really wrap my head around how I would use it.

Any ideas?

**Comment by arthurcvm** [comment]
Hi @faroutchris!
Please look the solution of [this discussion](https://github.com/adonisjs/core/discussions/1587#discussioncomment-77445).
